public/userInfo.php

```diff
@@ -502,113 +502,34 @@ function resize() {
-                    if ($scoreEarned > $scoreEarnedHardcore) {
-                        $scoreEarnedSoftcore = $scoreEarned - $scoreEarnedHardcore;
-                        echo ", $scoreEarnedSoftcore softcore points";
-                    }
-                } elseif ($scoreEarned) {
-                    echo "$scoreEarned/$maxPossibleScore softcore points";
-                } else {
-                    echo "0/$maxPossibleScore points";
-                }
-            }
-            echo "</div>";
-
-            RenderGameProgress($numPossibleAchievements, $numAchieved - $numAchievedHardcore, $numAchievedHardcore, fullWidthUntil: 'md');
-            echo "</div>";
-
-            echo "<div class='mb-5'>";
-            if (isset($userMassData['RecentAchievements'][$gameID])) {
-                foreach ($userMassData['RecentAchievements'][$gameID] as $achID => $achData) {
-                    $badgeName = $achData['BadgeName'];
-                    $achID = $achData['ID'];
-                    $achPoints = $achData['Points'];
-                    $achTitle = $achData['Title'];
-                    $achDesc = $achData['Description'];
-                    $achHardcore = $achData['HardcoreAchieved'];
-
-                    $unlockedStr = "";
-                    $class = 'badgeimglarge';
-
-                    if (!$achData['IsAwarded']) {
-                        $badgeName .= "_lock";
-                    } else {
-                        $achUnlockDate = getNiceDate(strtotime($achData['DateAwarded']));
-                        $unlockedStr = "<br clear=all>Unlocked: $achUnlockDate";
-                        if ($achHardcore == 1) {
-                            $unlockedStr .= "<br>HARDCORE";
-                            $class = 'goldimage';
-                        }
-                        $achData['Unlock'] = $unlockedStr;
-                    }
-
-                    echo achievementAvatar(
-                        $achData,
-                        label: false,
-                        icon: $badgeName,
-                        iconSize: 48,
-                        iconClass: $class,
-                    );
-                }
-            }
+    if ($user) {
+        echo "<div class='my-8'>";
+        echo Blade::render('
+            <x-user-recently-played
+                :recentlyPlayedCount="$recentlyPlayedCount"
+                :recentlyPlayedEntities="$recentlyPlayedEntities"
+                :recentAchievementEntities="$recentAchievementEntities"
+                :recentAwardedEntities="$recentAwardedEntities"
+                :targetUsername="$targetUsername"
+                :userAwards="$userAwards"
+            />
+        ', [
+            'recentlyPlayedCount' => $userMassData['RecentlyPlayedCount'],
+            'recentlyPlayedEntities' => $userMassData['RecentlyPlayed'],
+            'recentAchievementEntities' => $userMassData['RecentAchievements'],
```

Comment:
I get an exception on this line accessing the profile page for the most recently registered user.

Reply:
Good catch, fixed in https://github.com/RetroAchievements/RAWeb/pull/1989/commits/243254d177ab6b39570841da23932f30ebec9ab6 by setting some sane fallbacks.